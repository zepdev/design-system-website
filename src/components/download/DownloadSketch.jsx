import React from 'react'
import withStyles from 'react-jss'
import classnames from 'classnames'
import ButtonBase from '../button/ButtonBase'
import DownloadIcon from '../icons/DownloadIcon'
import sketchImg from '../../assets/images/sketch-symbol.svg'

const styles = theme => ({
  root: {
    marginBottom: `${ theme.spacing.l.rem * 2 }rem`,
    background: theme.colors.gray.grayLighter.hex,
    padding: `${ theme.spacing.l.rem }rem`,
    flexBasis: '40%',
  },
  fixedWidth: {
    width: '100%',
  },
  [`@media (min-width: ${ theme.breakpoints.s })`]: {
    fixedWidth: {
      width: 350,
    },
  },
  container: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  text: {
    marginBottom: `${ theme.spacing.xl.rem * 2 }rem`,
  },
  trademark: {
    marginBottom: 5,
    textTransform: 'capitalize',
    fontSize: 10,
  },
  img: {
    width: 46,
    height: 46,
  },
  button: {
    width: 46,
    height: 46,
    background: theme.colors.gray.white.hex,
    padding: `${ theme.spacing.xs.rem }rem`,
  },
  icon: {
    color: theme.colors.gray.grayLight.hex,
  },
})

const DownloadSketch = ({ demo, classes }) => {
  // const handleDownload = () => {
  //   let myHeaders = new Headers({
  //     'Content-Type': 'text/css',
  //   })
  //   fetch(
  //     'https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/css/zeppelin-element-library.css',
  //     {
  //       headers: myHeaders,
  //     }
  //   )
  //     .then(function(response) {
  //       console.log(response.text())
  //       return response.json()
  //     })
  //     .then(function(json) {
  //       console.log(json)
  //     })
  //     .catch(function(err) {
  //       console.log('Fetch problem: ' + err.message)
  //     })
  //   const filename = 'zeppelin-element-library.css'
  // const blob = new Blob([data], { type: 'application/octet-stream' })
  // if (typeof window.navigator.msSaveBlob !== 'undefined') {
  //   // IE workaround for "HTML7007: One or more blob URLs were
  //   // revoked by closing the blob for which they were created.
  //   // These URLs will no longer resolve as the data backing
  //   // the URL has been freed."
  //   window.navigator.msSaveBlob(blob, filename)
  // } else {
  //   let blobURL = window.URL.createObjectURL(blob)
  //   let tempLink = document.createElement('a')
  //   tempLink.style.display = 'none'
  //   tempLink.href = blobURL
  //   tempLink.setAttribute('download', filename)

  //   // Safari thinks _blank anchor are pop ups. We only want to set _blank
  //   // target if the browser does not support the HTML5 download attribute.
  //   // This allows you to download files in desktop safari if pop up blocking
  //   // is enabled.
  //   if (typeof tempLink.download === 'undefined') {
  //     tempLink.setAttribute('target', '_blank')
  //   }

  //   document.body.appendChild(tempLink)
  //   tempLink.click()
  //   document.body.removeChild(tempLink)
  //   window.URL.revokeObjectURL(blobURL)
  // }
  // }
  return (
    <div
      className={classnames(classes.root, {
        [classes.fixedWidth]: demo,
      })}
    >
      <p className={classnames('zep-typo--normal-3', classes.text)}>
        Color palettes for Sketch <sup className={classes.trademark}>TM</sup>
      </p>
      <div className={classes.container}>
        <img src={sketchImg} alt="sketch logo" className={classes.img} />
        <ButtonBase
          // onClick={handleDownload}
          className={classes.button}
          disabled={demo}
          target="_blank"
          href="https://s3.eu-central-1.amazonaws.com/com.zeppelin.zds.assets/zel/css/zeppelin-element-library.css"
          download="zeppelin-element-library.css"
          aria-label="download sketch file"
        >
          <DownloadIcon className={classes.icon} />
        </ButtonBase>
      </div>
    </div>
  )
}

export default withStyles(styles)(DownloadSketch)
