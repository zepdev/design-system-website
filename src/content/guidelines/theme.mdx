---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
