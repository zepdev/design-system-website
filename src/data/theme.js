module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
