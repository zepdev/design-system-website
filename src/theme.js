import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
