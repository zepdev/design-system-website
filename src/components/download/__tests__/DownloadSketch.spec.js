import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from '../../../data/theme'
import DownloadSketch from '../DownloadSketch'

describe('DownloadSketch', () => {
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <DownloadSketch />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
