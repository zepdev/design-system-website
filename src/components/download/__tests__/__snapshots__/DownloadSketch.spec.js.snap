// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadSketch renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="DownloadSketch-root-0-1-1"
      >
        <p
          class="zep-typo--normal-3 DownloadSketch-text-0-1-4"
        >
          Color palettes for Sketch 
          <sup
            class="DownloadSketch-trademark-0-1-5"
          >
            TM
          </sup>
        </p>
        <div
          class="DownloadSketch-container-0-1-3"
        >
          <img
            alt="sketch logo"
            class="DownloadSketch-img-0-1-6"
            src="test-file-stub"
          />
          <a
            aria-label="download sketch file"
            class="ButtonBase-button-0-1-9 DownloadSketch-button-0-1-7"
            download="zeppelin-element-library.css"
            href="https://cdn-zel.zepdev.net/zel/css/zeppelin-element-library.css"
            target="_blank"
          >
            <svg
              aria-labelledby="title"
              class="DownloadSketch-icon-0-1-8"
              height="32"
              viewBox="0 0 24 24"
              width="32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_copy"
              >
                Download Icon
              </title>
              <path
                d="M16 13h-3V3h-2v10H8l4 4 4-4zM4 19v2h16v-2H4z"
                fill="currentColor"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="DownloadSketch-root-0-1-1"
    >
      <p
        class="zep-typo--normal-3 DownloadSketch-text-0-1-4"
      >
        Color palettes for Sketch 
        <sup
          class="DownloadSketch-trademark-0-1-5"
        >
          TM
        </sup>
      </p>
      <div
        class="DownloadSketch-container-0-1-3"
      >
        <img
          alt="sketch logo"
          class="DownloadSketch-img-0-1-6"
          src="test-file-stub"
        />
        <a
          aria-label="download sketch file"
          class="ButtonBase-button-0-1-9 DownloadSketch-button-0-1-7"
          download="zeppelin-element-library.css"
          href="https://cdn-zel.zepdev.net/zel/css/zeppelin-element-library.css"
          target="_blank"
        >
          <svg
            aria-labelledby="title"
            class="DownloadSketch-icon-0-1-8"
            height="32"
            viewBox="0 0 24 24"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="title_copy"
            >
              Download Icon
            </title>
            <path
              d="M16 13h-3V3h-2v10H8l4 4 4-4zM4 19v2h16v-2H4z"
              fill="currentColor"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
